src/extension_commands.ml

```
@@ -534,6 +534,143 @@ module Copy_type_under_cursor = struct
+            let position =
+              TextEditor.selection text_editor |> Selection.active
+            in
+            let* query_input =
+              Vscode.(
+                Window.showInputBox
+                  ~options:
+                    (InputBoxOptions.create
+                       ~title:"Search By Type"
+                       ~prompt:
+                         "Perform a search by type request by providing a type \
+                          signature to look for"
+                       ~placeHolder:"int -> string"
+                       ~value:"int -> string"
+                       ~password:false
+                       ~ignoreFocusOut:true
+                       ())
+                  ())
+            in
+            match query_input with
+            | Some query -> (
+              let* type_search_results =
+                get_search_results
+                  ~query
+                  ~with_doc:true
+                  ~limit:100
+                  ~position
+                  text_editor
+                  client
+              in
+              let* type_picker = display_search_results type_search_results in
+              match type_picker with
+              | Some text ->
+                let+ type_inserted =
+                  TextEditor.edit
+                    text_editor
+                    ~callback:(fun ~editBuilder ->
+                      TextEditorEdit.insert
+                        editBuilder
+                        ~location:position
+                        ~value:text)
+                    ()
+                in
+                if not type_inserted then
+                  show_message `Error "Unable to insert %s" text
+                else show_message `Info "Inserted %s" text
+              | None -> show_message `Info "No selection made" |> Promise.return
+              )
+            | _ ->
+              show_message `Error "Invalid input for type search"
+              |> Promise.return))
```

My guess is that this only happens if the user decides to cancel his actions, I don't think this is an error, we should just silently do nothing.